### PR TITLE
Add dark mode

### PR DESCRIPTION
**Pull request by nthiebes**

### CODE REVIEW

File: _source/atoms/headline/Headline.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ export default class Headline extends Component {
           display && 'headline--display',
           noMargin && 'headline--no-margin',
           color && `headline--color-${color}`,
+          darkMode && 'headline--dark-mode',
```
Comment (Esclaire) [inline _source/atoms/headline/Headline.js]: Can we merge `headline--color-${color}` and `headline--dark-mode` here?

File: _source/atoms/headline/Headline.scss [hunk 2]
```diff
@@ -11,6 +11,10 @@
   margin-top: 2rem;
   padding-bottom: 1rem;
 
+  &.headline--dark-mode {
+    color: #fff;
```
Comment (Esclaire) [inline _source/atoms/headline/Headline.scss]: You usually use variables for colors, does it make sense to stick to one design "system"?

File: _source/atoms/headline/HeadlineContainer.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { connect } from 'react-redux';
+import Component from './Headline';
+
+export const mapStateToProps = (state) => ({
+  darkMode: state.user.darkMode
+});
+
+const Container = connect(
+  mapStateToProps
+)(Component);
```
Comment (Esclaire) [inline _source/atoms/headline/HeadlineContainer.js]: Could you explain why we need to do it in two steps here? As Container just has mapStateProps, why do we need it at all?

File: _source/atoms/icon/IconContainer.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { connect } from 'react-redux';
+import Component from './Icon';
+
+export const mapStateToProps = (state) => ({
+  darkMode: state.user.darkMode
+});
```
Comment (Esclaire) [inline _source/atoms/icon/IconContainer.js]: Just now that I see this again, the dark mode basically just changes colors as all the other color-picker does, doesn't it? Why do we need to touch basic components to make it happen? What is the difference between dark mode and color changes?

File: _source/atoms/label/Label.scss [hunk 5]
```diff
@@ -3,4 +3,9 @@
   margin-top: 1rem;
   padding-bottom: 0.5rem;
   line-height: 1.556rem;
+  color: $color__font--dark;
+}
+
+.label--dark {
+  color: #fff;
```
Comment (Esclaire) [inline _source/atoms/label/Label.scss]: Somehow it feels confusing that `.label--dark` has the font color white 😆 Is there a possibility that other devs will work on this? Maybe we should document it somehow?

File: _source/atoms/paragraph/Paragraph.scss [hunk 6]
```diff
@@ -3,8 +3,12 @@
   font-family: $font__family--text;
   line-height: 1.556rem;
   padding-bottom: 1rem;
+}
+
+.paragraph--first {
```
Comment (Esclaire) [inline _source/atoms/paragraph/Paragraph.scss]: Is this the `:first-child` of a `<p>`?

File: _source/molecules/color-picker/ColorPicker.scss [hunk 7]
```diff
@@ -100,4 +112,5 @@
 .color-picker__checkbox {
   opacity: 0;
   position: absolute;
+  pointer-events: none;
```
Comment (Esclaire) [inline _source/molecules/color-picker/ColorPicker.scss]: Remember the issue I had, test it properly on windows devices 🦆

File: _source/molecules/empty/Empty.scss [hunk 8]
```diff
@@ -6,12 +6,21 @@
...
   max-width: 350px;
   text-align: center;
+  color: $color__font--dark;
 
   @include mobile {
     padding: 0 calc(3% + 6px);
   }
 }
 
+.empty--dark-mode {
```
Comment (Esclaire) [inline _source/molecules/empty/Empty.scss]: Can we keep it consistent? Sometimes the modifier for the dark-mode is dark, sometimes it's dark-mode (which I prefer, especially if "dark" is white).

File: _source/organisms/dashboards/Dashboards.scss [hunk 9]
```diff
@@ -118,6 +132,14 @@
...
         cursor: default;
       }
+
+      &.dashboards__item--dark-mode {
+        background-color: inherit;
+
+        .dashboards__label {
+          color: #fff;
+        }
+      }
```
Comment (Esclaire) [inline _source/organisms/dashboards/Dashboards.scss]: Just wondering, it's not possible to also use BEM here, right? Else you'd do that?

File: _source/organisms/modal/Modal.scss [hunk 10]
```diff
@@ -12,7 +12,7 @@
   align-items: center;
   transition: width 0s 0.2s linear;
 
-  &:before {
+  &::before {
```
Comment (Esclaire) [inline _source/organisms/modal/Modal.scss]: Consistency? :D U know that this is not wrong, tho?

File: _source/organisms/modal/modals/AddBookmark.js [hunk 11]
```diff
@@ -20,15 +20,6 @@ class AddBookmark extends Component {
...
 
-  // componentWillReceiveProps(nextProps) {
-  //   this.setState({
-  //     name: '',
-  //     url: '',
-  //     categoryId: nextProps.data.source === 'header' ? nextProps.data.categories[0].id : nextProps.data.categoryId,
-  //     value: 0
-  //   });
-  // }
-
```
Comment (Esclaire) [inline _source/organisms/modal/modals/AddBookmark.js]: 🗑 🎉

File: _source/templates/booky/config.scss [hunk 12]
```diff
@@ -8,9 +8,10 @@ $color--dark: #3e3e3e;
 
 $color__background--primary: $color--primary;
 $color__background--dark: #2d2c2c;
-$color__background--medium: #3e3e3e;
+$color__background--medium: $color--dark;
 $color__background--ligth: #f5f5f5;
 $color__background--white: #fff;
```
Comment (Esclaire) [inline _source/templates/booky/config.scss]: AHA! There it iiiiis, so you are also using a variable for basic colors, sooooo u can make use of this powerful weapon in your code as well! 🗡

File: _source/templates/booky/config.scss [hunk 13]
```diff
@@ -26,7 +27,7 @@ $color--4: #eb65a0;
 $color--5: #9b539c;
 $color--6: #29aba4;
 $color--7: #657786;
-$color--8: #555;
+$color--8: $color--dark;
```
Comment (Esclaire) [inline _source/templates/booky/config.scss]: Is there a deeper meaning to this numbers, is it self-explaining? Or can we give it a bit more love? ❤️

File: _source/atoms/headline/Headline.scss (hunk 2 above)
Comment (nthiebes) [reply _source/atoms/headline/Headline.scss]: Will do 👍

File: _source/atoms/headline/Headline.js (hunk 1 above)
Comment (nthiebes) [reply _source/atoms/headline/Headline.js]: I'll check!

File: _source/_assets/i18n/de.json [hunk 14]
```diff
@@ -185,6 +185,7 @@
   "customize.blurEffect": "Fenster mit einem Verwisch- und Graustufen-Hintergrundeffekt öffnen",
   "customize.stickyHeader": "Feste Kopfzeilen-Position",
   "customize.stickyToolbar": "Feste Funktionsleisten-Position",
+  "customize.darkMode": "Dunkler Modus",
```
Comment (Esclaire) [inline _source/_assets/i18n/de.json]:
```suggestion
  "customize.darkMode": "Dunkles Design",
```

File: _source/atoms/label/Label.scss (hunk 5 above)
Comment (Esclaire) [reply _source/atoms/label/Label.scss]: In other cases you use `--dark-mode` as modifier, might be also the right choice here?

File: _source/atoms/paragraph/Paragraph.scss (hunk 6 above)
Comment (nthiebes) [reply _source/atoms/paragraph/Paragraph.scss]: Nope, it's a prop to add some more space to the top of a paragraph. It's useful for a first paragraph.

File: _source/organisms/dashboards/Dashboards.scss (hunk 9 above)
Comment (nthiebes) [reply _source/organisms/dashboards/Dashboards.scss]: For the variable?

File: _source/organisms/modal/Modal.scss (hunk 10 above)
Comment (nthiebes) [reply _source/organisms/modal/Modal.scss]: It is consistent, I'm usually using `::` for pseudo-elements.

File: _source/templates/booky/config.scss (hunk 12 above)
Comment (nthiebes) [reply _source/templates/booky/config.scss]: 🛡

File: _source/templates/booky/config.scss (hunk 13 above)
Comment (nthiebes) [reply _source/templates/booky/config.scss]: They are colors that a user can choose, will rename them.

**Pull request merged**